tests/cypress/integration/api/apiDescription.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { apolloClient } from '../../support/apollo'
+import gql from 'graphql-tag'
+
+describe('Test if every type in graphQL API has description', () => {
+    it('Check every input for the User Type', async function () {
+        const noDesc = new Set()
+        await executeTest('adminQuery', noDesc)
+        console.log(noDesc)
+        expect(JSON.stringify(Array.from(noDesc))).to.equals('[]')
+    })
+})
+// Test to go down the AST of GraphQL to check for descriptions
+const executeTest = async (typeName, noDesc) => {
```

Comment:
Would greatly appreciate some elaboration on what are we testing here?
As GraphQL API is more dynamic and our requests/responses are never static (not sure if this is the best approach)

Reply:
this is an example of how the nested object of the graphql can be traversed.
It's expected to be removed once you add some of your own tests here.
as for GraphQL API not being static, i'm not exactly sure if you refer to production? because in test we know what we can expect in the response

Reply:
I spoken to @Fgerthoffert 
This test is actually very specific for the GraphQL end-point implemented by their team on personal token API (As they got a more "static" definition of GraphQL nodes)
If we are using JCR GraphQL end-point this will fail as the definition is much different

Reply:
And to add on this, the jcr node should really be used for the edition of content (for things dynamic that cannot be structured in a schema). For everything related to features provided by a module (admin tasks for example), or specific features accessing a known path, we should create dedicated nodes (and using the jcr should be an exception).

And in that context, as soon as you create new GraphQL nodes, then we want to ensure that we have descriptions for these, and this test becomes very relevant/necessary.